src/text.h

```diff
@@ -410,10 +410,12 @@ struct Text {
     }
 
     void ReplaceStr(const wxString &str) {
-        for (int i = 0, j; (j = t.Mid(i).Lower().Find(sys->searchstring)) >= 0;) {
+        wxString text = sys->casesensitivesearch ? t : t.Lower();

```

Comment:
This copies the entire string even in the case `Lower` does not need to be called. Also, the search string can be converted just once, at the start of the search, rather than for reach cell.